model/lifecycle.go

```diff
@@ -1425,15 +1425,17 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 
 	taskIds := []string{}
 	for _, b := range builds {
-		// Find the set of task names that already exist for the given build
-		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey))
+		// Find the set of tasks that already exist for the set we want to add
+		taskNames := append(pairs.ExecTasks.TaskNames(b.BuildVariant), pairs.DisplayTasks.TaskNames(b.BuildVariant)...)
+		tasksInBuild, err := task.Find(task.ByBuildIdAndTaskNames(b.Id, taskNames))
```

Comment:
There isn't an index with great cardinality with this query, so we'd be asking the database to do filtering without an applicable index in a for loop. This concerns me a bit. Is this change essential for the fix? An alternative if you want to keep track of which task names are in pairs would be to create another map.

Given though that that `SetActiveState` requires a pointer, I'm not sure if there's a better way to do this.

Reply:
I am surprised that build ID isn't a good enough index though.

Reply:
Previously we were getting all task names from a given build. Here we're asking the database to filter that list. There are potentially thousands of these, and we'd do it across many builds.

Reply:
Gotcha, then I'm just going to revert the last two commits: I'm happy with the solution as it was before.